src/Modules/UngradedModules.jl

```diff
@@ -8712,3 +8716,258 @@ function dual(f::ModuleFPHom{<:ModuleFP, <:ModuleFP, Nothing}; # Third parameter
+
+If ``M`` happens to be finite-dimensional as a ``k``-vectorspace, this returns its dimension; otherwise it throws an error.
+
+# Examples:
+```jldoctest
+julia> R,(x,y,z,w) = QQ["x","y","z","w"];
+
+julia> F = free_module(R,2);
+
+julia> M,_ = quo(F,[1*gen(F,1),x^2*gen(F,2),y^3*gen(F,2),z*gen(F,2),w*gen(F,2)]);
+
+julia> vector_space_dimension(M,1)
+2
+
+julia> vector_space_dimension(M,2)
+2
+
+julia> vector_space_dimension(M,3)
+1
+
+julia> vector_space_dimension(M)
+6
+
+```
+"""
+function vector_space_dimension(M::SubquoModule)
+  
+  R = base_ring(M)
+  F = ambient_free_module(M)
+  Mq,_ = sub(F,rels(M))
+
+  ambient_representatives_generators(M) == gens(F) || error("not implemented for M/N with non-trivial M")
+
+  o = default_ordering(M)
+  LM = leading_module(Mq,o)
+
+  has_monomials_on_all_axes(LM) || error("not a finite dimensional vector space")
+  
+  d = 0
+  sum_dim = 0
+  tempdim = vector_space_dimension(M,0)
+
+  while tempdim > 0
+    sum_dim = sum_dim + tempdim
+    d = d+1
+    tempdim = vector_space_dimension(M,d)
+  end
+ 
+  return sum_dim
+end
+
+function vector_space_dimension(M::SubquoModule,d::Int64)
+  R = base_ring(M)
+  F = ambient_free_module(M)
+  Mq,_ = sub(F,rels(M))
+
+  ambient_representatives_generators(M) == gens(F) || error("not implemented for M/N with non-trivial M")
+
+  o = default_ordering(M)
+  LM = leading_module(Mq,o)
+
+  return length([x*e for x in all_monomials(R, d) for e in gens(F) if !(x*e in LM)])
+end
+  
+function vector_space_dimension(M::SubquoModule{T}
+  ) where {T<:MPolyLocRingElem{<:Field, <:FieldElem, <:MPolyRing, <:MPolyRingElem, 
+                               <:MPolyComplementOfKPointIdeal}}
+  F = ambient_free_module(M)
+  Mq,_ = sub(F,rels(M))
+
+  M_shift,_,_ = shifted_module(Mq)
+  o = negdegrevlex(base_ring(M_shift))*lex(ambient_free_module(M_shift))
+  LM = leading_module(M_shift,o)
+  return vector_space_dimension(quo(ambient_free_module(LM),gens(LM))[1])
+end
+
+function vector_space_dimension(M::SubquoModule{T},d::Int64
+  ) where {T<:MPolyLocRingElem{<:Field, <:FieldElem, <:MPolyRing, <:MPolyRingElem, 
+                               <:MPolyComplementOfKPointIdeal}}
+  F = ambient_free_module(M)
+  Mq,_ = sub(F,rels(M))
+
+  M_shift,_,_ = shifted_module(Mq)
+  o = negdegrevlex(base_ring(M_shift))*lex(ambient_free_module(M_shift))
+  LM = leading_module(M_shift,o)
+  return vector_space_dimension(quo(ambient_free_module(LM),gens(LM))[1],d)
+end
+
+function vector_space_dimension(M::SubquoModule{T}
+  ) where {T<:MPolyLocRingElem}
+  error("only available in global case and for localization at a point")
+end
+
+function vector_space_dimension(M::SubquoModule{T},d::Int64
+  ) where {T<:MPolyLocRingElem}
+  error("only available in global case and for localization at a point")
+end
+
+@doc raw"""
+    vector_space_basis(M::SubquoModule, d::Int)
+
+Let ``R`` be a `MPolyAnyRing` over a field ``kk`` and let ``M`` be a subquotient module over ``R``.
```

Comment:
```suggestion
Let ``R`` be a `MPolyAnyRing` over a field ``kk`` and let ``M`` be a subquotient module over ``R``.
```
Should be ``k`` or $\mathbb{k}$ as we would write in an article. I would say that `kk` is for code only.